File: platform/android/MapboxGLAndroidSDK/src/main/java/org/maplibre/android/location/MapLibreLatLngAnimator.java
package org.maplibre.android.location;

import android.animation.TypeEvaluator;

import androidx.annotation.NonNull;

import org.maplibre.android.geometry.LatLng;

class MapLibreLatLngAnimator extends MapLibreAnimator<LatLng> {

  MapLibreLatLngAnimator(@NonNull LatLng[] values, @NonNull AnimationsValueChangeListener updateListener,
                         int maxAnimationFps) {
    super(values, updateListener, maxAnimationFps);
  }

  @NonNull
  @Override
  TypeEvaluator provideEvaluator() {
    return new LatLngEvaluator();
  }
}

File: platform/android/MapLibreAndroid/src/main/java/org/maplibre/android/location/MapLibreLatLngAnimator.kt
package org.maplibre.android.location

import android.animation.TypeEvaluator
import org.maplibre.android.geometry.LatLng
import org.maplibre.android.location.MapLibreAnimator.AnimationsValueChangeListener

internal class MapLibreLatLngAnimator(
    values: Array<LatLng>,
    updateListener: AnimationsValueChangeListener<LatLng>,
    maxAnimationFps: Int,
) : MapLibreAnimator<LatLng>(values, updateListener, maxAnimationFps) {
    override fun provideEvaluator(): TypeEvaluator<*> = LatLngEvaluator()
}
